middleware/markdown/markdown.go

```diff
@@ -99,74 +70,67 @@ func (md Markdown) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 			continue
 		}
 
+		var dirents []os.FileInfo
 		fpath := r.URL.Path
 		if idx, ok := middleware.IndexFile(md.FileSys, fpath, md.IndexFiles); ok {
+			// We're serving a directory index file, which may be a markdown
+			// file with a template.  Let's grab a list of files this directory
+			// URL points to, and pass that in to any possible template invocations,
+			// so that templates can customize the look and feel of a directory.
+
+			fdp, err := md.FileSys.Open(fpath)
+			if err != nil {
+				return http.StatusInternalServerError, err
+			}
+			dirents, err = fdp.Readdir(-1)
+			if err != nil {
+				return http.StatusInternalServerError, err
+			}
```

Comment:
Done.  Ignoring the error on `Readdir(-1)` as well, a directory could be `--x--x--x`, which would prevent reading it, but it could still be useful to have a template render something.